src/item.cpp

```diff
@@ -6921,6 +6921,16 @@ bool item::is_reloadable_helper( const itype_id &ammo, bool now ) const
         return true;
     }
 
+    if( is_watertight_container() && !contents.empty() &&
+        ( contents.legacy_front().typeId() == ammo ) ) {
+        return true;
+    }
+
+    // Could be more robust. Assumes ammo is liquid - cannot determine from itype_id.
```

Comment:
You can use `Item_factory::find_template` to get the `itype` and check `itype::phase` for item phase.

Reply:
Good suggestion. I have the ammo phase being checked now. It should help make the code a little more future proof.

Reply:
Did you forgot to push the code? Because the code here is not changed at all.

Reply:
Yes. Updated now.